src/source/Sctp/Sctp.c

```diff
@@ -170,6 +170,38 @@ STATUS sctpSessionWriteMessage(PSctpSession pSctpSession, UINT32 streamId, BOOL
     return retStatus;
 }
 
+STATUS sctpSessionWriteDCEP(PSctpSession pSctpSession, UINT32 streamId, PCHAR pChannelName, UINT32 pChannelNameLen, PRtcDataChannelInit pRtcDataChannelInit)
+{
+     UNUSED_PARAM(pRtcDataChannelInit);
+     ENTERS();
+     STATUS retStatus = STATUS_SUCCESS;
+     struct sctp_sendv_spa spa;
+     PBYTE pPacket = NULL;
+     UINT32 pPacketSize = SCTP_DCEP_HEADER_LENGTH + pChannelNameLen;
+     UINT16 channelLenNetworkOrder = htons(pChannelNameLen);
+
+     CHK(pSctpSession != NULL && pChannelName != NULL, STATUS_NULL_ARG);
+     CHK((pPacket = (PBYTE) MEMCALLOC(1, pPacketSize)) != NULL, STATUS_NOT_ENOUGH_MEMORY);
```

Comment:
How big this can be? If it's capped, it can be a stack based variable instead of heap allocation

Reply:
12 + 65535 + 65535

Are you thinking of VLA (or something else)? I don't know of any stack allocations that are portable + error handling is good.

Reply:
It can be 2 + 65535 + 65535